### ISSUE TITLE
Fix autoname output length

### ISSUE THREAD

**Issue by MrPetovan** [friendica/friendica#5066]
Since we started using unit tests, Travis builds have been randomly failing because of the `autoname()` test.

I finally got into it, and figured that for a given random string length parameter, the output could be shorter.

After running `autoname(16)` 10,000 times, here are the results:

Length | Count | Ratio
-- | -- | --
16 chars | 9000 | 90.0 %
15 chars | 950 | 09.5 %
14 chars (?!) | 50 | 00.5%

This is because we are stripping letters from the tail end of the string after it has been generated.

See 
https://github.com/friendica/friendica/blob/ce75177d4e40882ecffe35fcc7b101d8716801e0/include/text.php#L179-L188

Please assess if this trimming is warranted, and if yes suggest an alternative to guarantee the output length.

Thanks to @Rudloff for enabling tests, it already bears fruits.

**Comment by abanink** [comment]
The problem is that it is not well defined what `$len` is in the autoname() function. Currently it's a maximum, but the real length could be smaller indeed. You could end up with 14 chars if the intermediate result ends in 'qu': 'u' is stripped because of the first loop, then 'q' is stripped because of the explicit test for 'q'.

If we want a minimum of changes, we could
* shift the line `$word = substr($word,0,$len);` to the end of the function.
* make sure `$word` contains at least `$len` chars before the end of the function. That could be achieved by changing `for ($x = 0; $x < $len; $x ++)` to `for ($x = 0; $x < $len+2; $x ++)` as the maximum amount of chars that is being omitted is 2.

Then the result of autoname() should have exactly `$len` chars.

**Comment by abanink** [comment]
The problem is that it is not well defined what $len is in the autoname() function. Currently it's a maximum, but the real length could be smaller indeed. You could end up with 14 chars if the intermediate result ends in 'qu': 'u' is stripped because of the first loop, then 'q' is stripped because of the explicit test for 'q'.

I created a PR for this already, but looking at it again, I see that it defies the purpose of the 'noend' logic. I'll give it another try later - the PR should be rejected for now.

**Comment by abanink** [comment]
I created a second PR.
I tested for correct length & end chars using this script:

```
#!/usr/bin/php
<?php
require 'include/text.php';

$numRuns = 10000;
print "Running autoname $numRuns times\n";


$noend = ['bl', 'br', 'cl','cr','dr','fl','fr','gl','gr',
				'kh', 'kl','kr','mn','pl','pr','rh','tr','qu','wh','q'];

$output = array();
for ($i=0;$i<$numRuns;$i++)
{
	$myword = autoname(16);
	$len = strlen($myword);

	foreach ($noend as $noe) {
		$noelen = strlen($noe);
		if ((strlen($myword) > $noelen) && (substr($myword, -$noelen) == $noe)) {
			print "ERROR: Resulting generated word ($myword) ends in $noe!\n";
			break;
		}
	}

	if (array_key_exists($len,$output)) $output[$len]=$output[$len]+1; else $output[$len]=1; 

}

print "Result ";
print_r($output);
?>
```

Output:
```
Running autoname 10000 times
Result Array
(
    [16] => 10000
)
```

Using the original code, it would output
```
Running autoname 10000 times
Result Array
(
    [16] => 9047
    [15] => 907
    [14] => 46
)
```

**Comment by MrPetovan** [comment]
Closed via #5140
